test/functional/readconcern_tests.js

```diff
@@ -1,498 +1,361 @@
+    expect(info.commands.started[i]).to.have.property('commandName', commandName);
+    expect(info.commands.succeeded[i]).to.have.property('commandName', commandName);
+    if (level !== undefined) {
+      expect(info.commands.started[i]).to.have.nested.property('command.readConcern.level', level);
+    } else {
+      expect(info.commands.started[i].command.readConcern).to.equal(undefined);
     }
-  });
-
-  it('Should set majority readConcern using MongoClient with options', {
-    metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
-
-    test: function(done) {
-      var configuration = this.configuration;
-      var listener = require('../..').instrument(function(err) {
-        test.equal(null, err);
-      });
-
-      // Contains all the apm events
-      var started = [];
-      var url = configuration.url();
-      var options = {
-        readConcern: {
-          level: 'majority'
+  }
+
+  tests.forEach(test => {
+    it(test.description, {
+      metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
+
+      test: function(done) {
+        // Get a new instance
+        const configuration = this.configuration;
+        client = !test.urlOptions
+          ? configuration.newClient({ w: 1 }, { poolSize: 1, readConcern: { level: test.level } })
+          : null;
+
+        if (test.urlReadConcernLevel || test.urlOptions) {
+          url = configuration.url();
+          if (!test.urlOptions) {
+            url =
+              url.indexOf('?') !== -1
+                ? f('%s&%s', url, test.urlReadConcernLevel)
+                : f('%s?%s', url, test.urlReadConcernLevel);
+          } else {
+            const options = {
+              readConcern: {
+                level: test.level
+              }
+            };
+            client = configuration.newClient(url, options);
+          }
         }
-      };
 
-      // Connect using mongoclient
-      const client = configuration.newClient(url, options);
-      client.connect(function(err, client) {
-        expect(err).to.not.exist;
+        client.connect((err, client) => {
+          expect(err).to.not.exist;
 
-        var db = client.db(configuration.db);
-        test.deepEqual({ level: 'majority' }, db.s.readConcern);
+          const db = client.db(configuration.db);
+          expect(db.s.readConcern).to.deep.equal({ level: test.level });
 
-        // Get a collection
-        var collection = db.collection('readConcernCollection');
-        // Validate readConcern
-        test.deepEqual({ level: 'majority' }, collection.s.readConcern);
-        // Perform a find using the readConcern
-        listener.on('started', function(event) {
-          if (event.commandName === 'find') started.push(event);
-        });
+          // Get a collection
+          const collection = test.collectionReadConcernLevel
+            ? db.collection('readConcernCollection', {
+                readConcern: { level: test.collectionReadConcernLevel }
```

Comment:
Are there any cases where `test.collectionReadConcernLevel` is different from `test.level`? If not, we can eliminate `test.collectionReadConcernLevel` in favor of `test.level`.